### DISCUSSION TITLE
Apple Mail thinks 'Message is from a mailing list' for all email from AnonAddy

### DISCUSSION THREAD

**Discussion by ccorneli** [anonaddy/anonaddy#165] [Q&A] [answered]
All the email i get forwarded from AnonAddy is shown in Apple Mail client saying at the top:  _"This Message is from a mailing list"_ and it gives a link to unsubscribe.  I realize this is a 'feature' of Apple mail, but what is special about AnonAddy forwarding that causes apple mail to think that?  

Originally I thought it just looked for an 'unsubscribe' link in the message body, but my message only has 2 words in the body: 'test message'.   Then I thought it was from the banner saying it was alias created by AnonAddy, and click the link to block it.
However I disabled the banner and still see the message.  

Is there something special in the raw email headers which causes apple mail to think it is from a mailing list?

**Comment by willbrowningme** [comment]
This is because of the `List-Unsubscribe:` header that is added to each forwarded email from AnonAddy. This header contains an email address you can send to and also a link you can click that will deactivate the alias.

The reason this is added is because it can improve email deliverability.

**Reply by ccorneli** [reply]
Can you elaborate on the benefits? I'm not sure what you mean by 'improve email deliverability', but I don't see a reason to still having the header, if the banner was removed and there is nothing to click in the message body to deactivate the alias.

**Reply by willbrowningme** [reply]
Please see this article from Apple about best practices - [https://support.apple.com/en-us/HT204137](https://support.apple.com/en-us/HT204137)

Specifically:

> Offer an unsubscribe link, so that the recipient can unsubscribe immediately.

**Reply by ccorneli** [reply]
Thanks for the info, it's both interesting and useful.. however that is a section for 'Bulk Email'.   I think the application of it here is a bit misguided since most of my email is direct to me as individual, not a bulk email to a group, and as a result every email has that message unnecessarily.  (especially if Banner is disabled)

In fact, I think it actually conflicts with existing unsubscribe links. (although I haven't tested)

For example, when I get notified of your GitHub replies using my old forwarding service, the unsubscribe link points to Github which would take me to remove my interest in the topic.
But that same email sent via AA points the unsubscribe link to you which I expect disables the alias.

So my guess is that I would click the 'unsubscribe' provided by Apple, thinking I'm removing myself from GitHub, when in fact I'm actually asking you to block the alias.

Text from Github:

> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, _**view it on GitHub**_, or _**unsubscribe**_.


via my previous Email service:  (remove from watching the issue at Github)
```
List-Unsubscribe: <mailto:unsub+[TOKEN]@reply.github.com>
List-Unsubscribe: <https://github.com/notifications/unsubscribe/[TOKEN]>

```

via AnonAddy:  ( Block Github alias)
```
List-Unsubscribe: <mailto:[TOKEN]@unsubscribe.anonaddy.com?subject=unsubscribe>
List-Unsubscribe: <https://app.anonaddy.com/deactivate/[TOKEN]?signature=[TOKEN]>
```

Edit / Update:
I was able to give this a quick test using another alias.  I confirmed it deactivated the alias.  Not sure if you are also forwarding something to the source to further remove from their list, since any confirmation email would already be blocked.

**Reply by ccorneli** [reply]
@willbrowningme  -- Given the example above, this seems like a bug...  (or at least a feature request to exclude the headers if the banner is disabled).

**Comment by ccorneli** [comment, marked as answer]
I just found Issue #130 which discusses this issue.  (Hopefully it will get added to roadmap.)
